cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -96,39 +112,111 @@ public void addExecutionProposals(Collection<ExecutionProposal> proposals,
                                     Cluster cluster,
                                     ReplicaMovementStrategy replicaMovementStrategy) {
     LOG.trace("Cluster state before adding proposals: {}.", cluster);
-    maybeAddReplicaMovementTasks(proposals, cluster, replicaMovementStrategy);
+    maybeAddInterBrokerReplicaMovementTasks(proposals, cluster, replicaMovementStrategy);
+    maybeAddIntraBrokerReplicaMovementTasks(proposals);
     maybeAddLeaderChangeTasks(proposals, cluster);
+    sanityCheckExecutionTasks();
+    maybeDropReplicaSwapTasks();
   }
 
   /**
-   * For each proposal, create a replica action task if there is a need for moving replica(s) to reach expected final proposal state.
+   * Sanity check that if there is any intra-broker partition movement task generated, no inter-broker partition movement task
+   * is generated.
+   */
+  private void sanityCheckExecutionTasks() {
+    if (_remainingIntraBrokerReplicaMovements.size() > 0) {
+      for (ExecutionTask task : _remainingInterBrokerReplicaMovements) {
+        if (task.proposal().replicasToAdd().size() > 0) {
```

Comment:
because inter-broker replica movement also includes replica swap, here I want to check, if there is any intra-broker movement, then there can either be no inter-broker movement or only replica swap.